src/widgets/settingspages/HighlightingPage.cpp

```diff
@@ -209,73 +278,92 @@ HighlightingPage::HighlightingPage()
             {
-                view->getModel()->setData(clicked, selected,
-                                          Qt::DecorationRole);
-
-                if (fromMessagesTab)
+                /*
+                 * For preset highlights in the "Messages" tab, we need to
+                 * manually update the color map.
+                 */
+                auto instance = ColorProvider::instance();
+                switch (clicked.row())
                 {
-                    /*
-                     * For preset highlights in the "Messages" tab, we need to
-                     * manually update the color map.
-                     */
-                    auto instance = ColorProvider::instance();
-                    switch (clicked.row())
-                    {
-                        case 0:
-                            instance.updateColor(ColorType::SelfHighlight,
-                                                 selected);
-                            break;
-                        case 1:
-                            instance.updateColor(ColorType::Whisper, selected);
-                            break;
-                        case 2:
-                            instance.updateColor(ColorType::Subscription,
-                                                 selected);
-                            break;
-                    }
+                    case 0:
+                        instance.updateColor(ColorType::SelfHighlight,
+                                             selected);
+                        break;
+                    case 1:
+                        instance.updateColor(ColorType::Whisper, selected);
+                        break;
+                    case 2:
+                        instance.updateColor(ColorType::Subscription, selected);
+                        break;
                 }
             }
-        });
+        }
+    });
+}
+
+void HighlightingPage::tableCellClicked(const QModelIndex &clicked,
+                                        EditableModelView *view,
+                                        HighlightTab tab)
+{
+    if (tab == HighlightTab::Messages || tab == HighlightTab::Users)
+    {
+        using Column = HighlightModel::Column;
+        if (clicked.column() == Column::SoundPath)
+        {
+            this->openSoundDialog(clicked, view, Column::SoundPath);
+        }
+        else if (clicked.column() == Column::Color &&
+                 clicked.row() != HighlightModel::WHISPER_ROW)
+        {
+            this->openColorDialog(clicked, view, tab);
+        }
+    }
+    else if (tab == HighlightTab::Badges)
+    {
+        using Column = BadgeHighlightModel::Column;
+        if (clicked.column() == Column::SoundPath)
+        {
+            this->openSoundDialog(clicked, view, Column::SoundPath);
+        }
+        else if (clicked.column() == Column::Color)
+        {
+            this->openColorDialog(clicked, view, tab);
+        }
```

Comment:
Can this be a switch(tab)-case statement instead? Maybe it's not necessary, but since we're using fancy enums instead of strings it might be worth looking into in case we're gonna add any other kind of highlights in the future.

Reply:
Fixed :ok_hand: